### ISSUE TITLE
parse_tree with decimal points and integer ranges

### ISSUE THREAD

**Issue by Shimorina** [EmilStenstrom/conllu#21]
Hello,

In the latest version (1.2.1), `parse_tree` doesn't work properly if a sentence contains ids with integer ranges or decimal points.

```
data_file = """
1	En	en	ADP	_	_	2	case	_	_
2	1872	1872	NUM	_	_	26	obl	_	SpaceAfter=No
3	,	,	PUNCT	_	_	26	punct	_	_
4	quand	quand	SCONJ	_	_	8	mark	_	_
5	les	le	DET	_	Definite=Def|Number=Plur|PronType=Art	6	det	_	_
6	Georgiens	Georgiens	PROPN	_	_	8	nsubj	_	_
7	ont	avoir	AUX	_	Mood=Ind|Number=Plur|Person=3|Tense=Pres|VerbForm=Fin	8	aux	_	_
8	repris	reprendre	VERB	_	Gender=Masc|Number=Sing|Tense=Past|VerbForm=Part	26	advcl	_	_
9	les	le	DET	_	Definite=Def|Gender=Fem|Number=Plur|PronType=Art	10	det	_	_
10	commandes	commande	NOUN	_	Gender=Fem|Number=Plur	8	obj	_	_
11-12	du	_	_	_	_	_	_	_	_
11	de	de	ADP	_	_	13	case	_	_
12	le	le	DET	_	Definite=Def|Gender=Masc|Number=Sing|PronType=Art	13	det	_	_
13	gouvernement	gouvernement	NOUN	_	Gender=Masc|Number=Sing	10	nmod	_	SpaceAfter=No
14	,	,	PUNCT	_	_	26	punct	_	_
15	Barnett	Barnett	PROPN	_	_	26	nsubj	_	_
16	(	(	PUNCT	_	_	20	punct	_	SpaceAfter=No
17	qui	qui	PRON	_	PronType=Rel	20	nsubj:pass	_	_
18	avait	avoir	AUX	_	Mood=Ind|Number=Sing|Person=3|Tense=Imp|VerbForm=Fin	20	aux	_	_
19	été	être	AUX	_	Gender=Masc|Number=Sing|Tense=Past|VerbForm=Part	20	aux:pass	_	_
20	réélu	réélire	VERB	_	Gender=Masc|Number=Sing|Tense=Past|VerbForm=Part	15	acl:relcl	_	_
21	par	par	ADP	_	_	23	case	_	_
22	ce	ce	DET	_	Gender=Masc|Number=Sing|PronType=Dem	23	det	_	_
23	point	point	NOUN	_	Gender=Masc|Number=Sing	20	obl	_	SpaceAfter=No
24	)	)	PUNCT	_	_	20	punct	_	_
25	a	avoir	AUX	_	Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin	26	aux	_	_
26	rapporté	rapporter	VERB	_	Gender=Masc|Number=Sing|Tense=Past|VerbForm=Part	0	root	_	_
27	le	le	DET	_	Definite=Def|Gender=Masc|Number=Sing|PronType=Art	28	det	_	_
28	sceau	sceau	PROPN	_	_	26	obj	_	SpaceAfter=No
29	.	.	PUNCT	_	_	26	punct	_	_
"""
sentence_tree = parse_tree(data_file)
print(sentence_tree)
```
outputs 
```
[TokenTree<token={id=(11, '-', 12), form=du}, children=None>]
```

**Comment by EmilStenstrom** [comment]
Hi! This is a change from how the previous version of conllu worked, and not something that I consider a bug. What would you expect to get as an id in the case of a range or decimal point in the ID?

Here's a testcase that ensures I don't change this behaviour in the future: https://github.com/EmilStenstrom/conllu/blob/master/tests/test_parser.py#L145

**Comment by Shimorina** [comment]
Thanks for your answer.

My point is that when you do `parse_tree` from that sentence, I would expect to get the tree root with its children: `TokenTree<token={id=26, form=rapporté}, children=[...]>`, but it returns only one node without any children. That's the usual behaviour if you feed a sentence with normal ids.

I'm completely okay how those ids are treated if you do a simple `parse()`.

**Comment by EmilStenstrom** [comment]
Oh, I see what you mean! Great find, that's definitely something that I'll have to fix.

**Comment by EmilStenstrom** [comment]
Ah, found a simple fix. I simply filter out all the range and decimal nodes before building the tree. I'm guessing people building a tree are uninterested in those nodes anyway. 

I've released a new version to PyPI which includes this fix, 1.2.2: https://pypi.org/project/conllu/

Thanks again for your reporting this bug, appreciate it!

**Comment by Shimorina** [comment]
Thanks for handling the problem with the incredible speed!

I agree that those additional nodes can be ignored while building a tree.
Tack så mycket!
